src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ConfigFileBuildWrapper.java

```diff
@@ -57,22 +54,22 @@ public ConfigFileBuildWrapper(List<ManagedFile> managedFiles) {
         this.managedFiles = managedFiles;
     }
 
-    @Override
-    public Environment setUp(@SuppressWarnings("rawtypes") AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
-
-        final PrintStream logger = listener.getLogger();
-
-        if (build.getWorkspace() == null) {
-            throw new IllegalStateException("the workspace does not yet exist, can't provision config files - maybe slave is offline?");
+    @Override public void setUp(Context context, Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener, EnvVars initialEnvironment) throws IOException, InterruptedException {
+        final Map<ManagedFile, FilePath> file2Path = ManagedFileUtil.provisionConfigFiles(managedFiles, build, workspace, listener);
+        context.env = new EnvVars();
+        List<String> tempFiles = new ArrayList<String>();
+        for (Map.Entry<ManagedFile, FilePath> entry : file2Path.entrySet()) {
+            ManagedFile mf = entry.getKey();
+            FilePath fp = entry.getValue();
+            if (!StringUtils.isBlank(mf.variable)) {
+                context.env.put(mf.variable, fp.getRemote());
+            }
+            boolean noTargetGiven = StringUtils.isBlank(entry.getKey().targetLocation);
+            if (noTargetGiven) {
+                tempFiles.add(entry.getValue().getRemote());
+            }
         }
-
-        final Map<ManagedFile, FilePath> file2Path = ManagedFileUtil.provisionConfigFiles(managedFiles, build, listener);
-        // JENKINS-17555 this special env is required, as MavenModuleSetBuild only takes Environments from BuildWrapper into account, but not those from Actions registered by them.
-        final ManagedFilesEnvironment env = new ManagedFilesEnvironment(file2Path);
-        // Temporarily attach info about the files to be deleted to the build - this action gets removed from the build again by 'org.jenkinsci.plugins.configfiles.common.CleanTempFilesRunListener'
-        build.addAction(new CleanTempFilesAction(file2Path));
```

Comment:
This idiom cannot be used in a workflow, and the original code here was a misuse of `BuildWrapper`; should have been taking advantage of `Environment.tearDown`.